src/expr.rs

```diff
@@ -2481,6 +2482,26 @@ fn struct_lit_can_be_aligned(fields: &[ast::Field], base: &Option<&ast::Expr>) -
     fields.iter().all(|field| !field.is_shorthand)
 }
 
+#[allow(dead_code)]
+/// Check if a struct representation ends with comma by match on the chars of struct
+/// snippet reversely, once a FullCodeCharKind::Normal comma matched, returns true
+fn struct_lit_ends_with_comma(context: &RewriteContext, span: Span) -> bool {
```

Comment:
I added a function to check if a struct lit ends with comma here, but I need to match the comma reversely because normally match might never handle the last field before it returns.

I had tried to `impl DoubleEndedIterator for CharClasses {...}`, the basic `next_back` method is easy to learn from std-docs, but it's a bit difficult for me to tackle the method on `CharClasses`, could you please give me some tips the deal with it? @topecongiro